core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/registry/MetricRegistry.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics.registry;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.metrics.meter.Counter;
+import software.amazon.awssdk.metrics.meter.Gauge;
+import software.amazon.awssdk.metrics.meter.Metric;
+import software.amazon.awssdk.metrics.meter.Timer;
+
+/**
+ * Registry to store the collected metrics data. The interface can be used to store metrics for ApiCall and ApiCallAttempt.
+ * For a ApiCall, there can be multiple attempts and so a MetricRegistry has the option to store other MetricRegistry instances.
+ */
+@SdkPublicApi
+public interface MetricRegistry {
+
+    /**
+     * Return the ApiCall level metrics registered in this metric registry as a map of metric name to metric instance.
+     * Only metrics that can be recorded once for entire request lifecycle are recorded here.
+     *
+     * The method does not return the Api Call Attempt metrics. For metrics recorded separately for each attempt,
+     * see {@link #apiCallAttemptMetrics()}.
+     */
+    Map<String, Metric> metrics();
+
+
+    /**
+     * Return an ordered list of {@link MetricRegistry} instances recorded for each Api Call Attempt in the request execution.
+     * Each Api call attempt metrics are recorded as a separate {@link MetricRegistry} instance in the given list.
+     *
```

Comment:
missing `<p>`. I think there are other places also missing the tag. Can you do a global check?

Reply:
Added in few classes that I noticed. Also realized the `<p>` don't need an end tag